pkgs/tools/misc/lookatme/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib, python3, fetchFromGitHub }:
+
+let
+  py = python3.override {
+    packageOverrides = self: super: {
+      self = py;
+      # use click 7
+      click = self.callPackage ../../../development/python2-modules/click/default.nix { };
```

Comment:
@AmeerTaweel first, welcome. :rocket:  Secondly, one thing to note is we're getting rid of python 2 (https://github.com/NixOS/nixpkgs/issues/148779) so i'd either:

create a click package that's v7 using python 3 (this might help: https://github.com/NixOS/nixpkgs/blob/6bdc4f3fe2ba3ed340b84b3b715013e07937c892/pkgs/development/python-modules/click/default.nix#L1)

or you'd need to update lookatme upstream to use click v8. Click v8 came out 2021-05-11 so it's probably about time :sweat_smile:

Reply:
I want to submit a pull-request to make upstream use `click v8`. This is my idea on how to do it:

+ Update `requirements.txt` of the package.
+ Update the Nix derivation.
+ Build and test.
+ Submit pull-request to upstream.
+ They (hopefully) merge it.
+ I submit the updated Nix derivation.

Is that good? Or am I missing some thing(s)?

Reply:
Of course, I'm assuming that doing the update will not break anything. If it breaks something, I will have to fix it.

Reply:
Yeah that sounds good. I'm not sure how `buildPythonApplication`/`fetchPypi` work exactly but you can probably change `src =` to be `src = ./your-repo-with-changes;`

I'd note a major version change from v7 to v8 probably has some breaking changes that'll need some modifications in the python code, especially since the requirements.txt explicitly says not v8 currently.

For submitting the updated nix derivation you could either pull the existing 2.3.2 version and patch it, or wait for 2.3.3 with your changes. (example of a patch fixing something ahead of a new release https://github.com/NixOS/nixpkgs/pull/158707/files#diff-e8c4f9d22def9371b17eb15d3f5df0dd00e177ff96deea05a265222880ec7cb6R15-R22)

Reply:
Thanks for your help. Will open an issue upstream to inform the authors about what I want to do.

Reply:
Made an [issue](https://github.com/d0c-s4vage/lookatme/issues/131) upstream and waiting for approval.

Reply:
@06kellyjac it's been about two months since I created the issue but got no response at all. What should I do?
Should I forget about that and try the first option of `click v7` with `python3`?

Reply:
The lookatme project doesn't seem to have made any movement in a long time.

The preferred solution would be to update lookatme. We can then package it with the patch file from the PR until that's merged in.

You could try maintain a copy of click v7 just for use by lookatme in nixpkgs but at some point we do cull old versions of software.

I think you'll need to review if lookatme is important enough for you to put in the effort to make this PR bumping to click v8.
It might even come down to you forking the project and maintaining it yourself since the main dev doesn't look active in any projects anymore

Reply:
Actually it's not that important to me. I use it for my presentations, but to be honest, it does not worth the effort...

Reply:
Yeah, makes sense.
Pip install or a virtual env might work. Or you can run it in a container